code/headers/joystick_value.hpp

```diff
@@ -1,6 +1,45 @@
 #pragma once
+#include <hwlib.hpp>
 
-struct joystick_value_s {
+class joystick_value_c {
+public:
     int x;
     int y;
+
+    // operators
+    bool operator!=(const joystick_value_c &rhs) {
+        return ((this->x != rhs.x) || (this->y != rhs.y));
```

Comment:
Functions should be defined in cpp, this applies to all the functions in this class (when changed to a struct this comment will become invalid)

Reply:
Changed to struct and move the operators to a separate .cpp